server/src/main/java/org/elasticsearch/search/MultiValueMode.java

```diff
@@ -63,16 +58,38 @@ protected long pick(SortedNumericDocValues values) throws IOException {
         }
 
         @Override
-        protected long pick(SortedNumericDocValues values, long missingValue, DocIdSetIterator docItr, int startDoc, int endDoc) throws IOException {
+        protected long pick(SortedNumericDocValues values, long missingValue, DocIdSetIterator docItr, int startDoc, int endDoc, Integer maxChildren) throws IOException {
```

Comment:
why not an `int` and make the option defaults to `Integer.MAX_VALUE` ?

Reply:
I could not decide for default value.

I will change it.